https://gitlab.xiph.org/xiph/speex/-/merge_requests/13

From b15993fd560c90658ff2e4974e8b3a1afcfcfff2 Mon Sep 17 00:00:00 2001
From: Andrei Slavoiu <ansla80@yahoo.com>
Date: Tue, 16 May 2023 04:49:45 +0100
Subject: [PATCH] Fix --enable-valgrind build

VALGRIND_MAKE_READABLE has been replaced with VALGRIND_MAKE_MEM_DEFINED
starting with version 3.2 (2006) https://valgrind.org/docs/manual/dist.news.old.html.

Bug: https://bugs.gentoo.org/906455
--- a/libspeex/nb_celp.c
+++ b/libspeex/nb_celp.c
@@ -172,7 +172,7 @@ void *nb_encoder_init(const SpeexMode *m)
    st->highpass_enabled = 1;
 
 #ifdef ENABLE_VALGRIND
-   VALGRIND_MAKE_READABLE(st, NB_ENC_STACK);
+   VALGRIND_MAKE_MEM_DEFINED(st, NB_ENC_STACK);
 #endif
    return st;
 }
@@ -1113,7 +1113,7 @@ void *nb_decoder_init(const SpeexMode *m)
    st->highpass_enabled = 1;
 
 #ifdef ENABLE_VALGRIND
-   VALGRIND_MAKE_READABLE(st, NB_DEC_STACK);
+   VALGRIND_MAKE_MEM_DEFINED(st, NB_DEC_STACK);
 #endif
    return st;
 }
--- a/libspeex/sb_celp.c
+++ b/libspeex/sb_celp.c
@@ -223,7 +223,7 @@ void *sb_encoder_init(const SpeexMode *m)
    speex_encoder_ctl(st->st_low, SPEEX_GET_SAMPLING_RATE, &st->sampling_rate);
    st->sampling_rate*=2;
 #ifdef ENABLE_VALGRIND
-   VALGRIND_MAKE_READABLE(st, (st->stack-(char*)st));
+   VALGRIND_MAKE_MEM_DEFINED(st, (st->stack-(char*)st));
 #endif
    return st;
 }
@@ -1017,7 +1017,7 @@ void *sb_decoder_init(const SpeexMode *m)
    st->seed = 1000;
 
 #ifdef ENABLE_VALGRIND
-   VALGRIND_MAKE_READABLE(st, (st->stack-(char*)st));
+   VALGRIND_MAKE_MEM_DEFINED(st, (st->stack-(char*)st));
 #endif
    return st;
 }
-- 
GitLab
